tools/dev/ensure_headers.py

```diff
@@ -0,0 +1,122 @@
+"""Ensure license and copyright statements are in source files.
+
+From https://www.bestpractices.dev/en/projects/7783?criteria_level=2:
+
+    The project MUST include a copyright statement in each source file, identifying the
+    copyright holder (e.g., the [project name] contributors). [copyright_per_file]
+    This MAY be done by including the following inside a comment near the beginning of
+    each file: "Copyright the [project name] contributors.".
+
+And:
+
+    The project MUST include a license statement in each source file.
+
+This script ensures that we use consistent license naming in consistent locations
+toward the top of each file.
+"""
+# License: BSD-3-Clause
+# Copyright the MNE-Python contributors.
+
+import re
+from pathlib import Path
+
+import numpy as np
+from git import Repo
+
+repo = Repo(Path(__file__).parents[2])
+
+LICENSE_LINE = "# License: BSD-3-Clause"
+COPYRIGHT_LINE = "# Copyright the MNE-Python contributors."
+
+
+def get_paths_from_tree(root, level=0):
+    for entry in root:
+        if entry.type == "tree":
+            for x in get_paths_from_tree(entry, level + 1):
+                yield x
+        else:
+            yield Path(entry.path)  # entry.type
+
+
+def _ensure_license(lines, path):
+    # 1. Keep existing
```

Comment:
@drammock this is the one new file. Might be easiest to check out my branch and look at `tools/dev/ensure_headers.py` locally. GitHub is really slow with a diff across so many files